projects/isy-angular-widgets/src/lib/validation/validation.ts

```diff
@@ -67,10 +72,39 @@ export class Validation {
       }
     }
 
-    const regexInputUnspecified =
-      /^(0{2}\.([0-1][1-2]|1[0-2])\.\d{4})$|^(0{2}\.0{2}\.\d{4})$|^(0{2}\.0{2}\.0{4})$|^(x{2}\.([0-1][1-2]|1[0-2])\.\d{4})$|^(x{2}\.x{2}\.\d{4})$|^(x{2}\.x{2}\.x{4})$/;
+    if (!(INPUT_UNSPECIFIED_REGEX.test(input) || isDateValid)) return {UNSPECIFIEDDATE: true};
 
-    if (!(RegExp(regexInputUnspecified).exec(input) !== null || isDateValid)) return {UNSPECIFIEDDATE: true};
+    return null;
+  }
+
+  /**
+   * Checks that the date is a valid unspecified date or valid date in german format DD.MM.YYYY.
+   * If the date in german format is not valid and not unspecified, a "DATE" error is thrown.
+   * E.g. unspecified dates: YYYY-MM-00, YYYY-00-00, 0000-00-00, YYYY-MM-xx, YYYY-xx-xx, xxxx-xx-xx
+   * For valid or valid unspecified dates, no error is thrown.
+   * @param c The control element the validator is appended to
+   * @returns The object {UNSPECIFIEDISODATE: true} if the validation fails; null otherwise
```

Comment:
Falsches Format in der Beschreibung; sollte ISO sein
```suggestion
  /**
   * Checks that the date is a valid unspecified date or valid date in ISO 8601 format YYYY-MM-DD.
   * If the date in ISO 8601 format is not valid and not unspecified, a "DATE" error is thrown.
   * E.g. unspecified dates: YYYY-MM-00, YYYY-00-00, 0000-00-00, YYYY-MM-xx, YYYY-xx-xx, xxxx-xx-xx
   * For valid or valid unspecified dates, no error is thrown.
   * @param c The control element the validator is appended to
   * @returns The object {UNSPECIFIEDISODATE: true} if the validation fails; null otherwise
```

Reply:
updated